charts/core/traefik/templates/_args.tpl

```diff
@@ -55,14 +55,17 @@ args:
   - "--providers.kubernetesingress.labelSelector={{ .Values.providers.kubernetesIngress.labelSelector }}"
   {{- end }}
   {{- end }}
-  {{- if and .Values.rbac.enabled .Values.rbac.namespaced }}
+  {{- if .Values.rbac.enabled }}
   {{- if .Values.providers.kubernetesCRD.enabled }}
   - "--providers.kubernetescrd.namespaces={{ template "providers.kubernetesCRD.namespaces" . }}"
   {{- end }}
   {{- if .Values.providers.kubernetesIngress.enabled }}
   - "--providers.kubernetesingress.namespaces={{ template "providers.kubernetesIngress.namespaces" . }}"
   {{- end }}
   {{- end }}
+  {{- if .Values.ingressClass.enabled }}
+  - "--providers.kubernetesingress.ingressclass="{{ .Release.Name }}
+  {{- end }}
```

Comment:
ingressclass is not necisarily releasename either.
It's a string.

Reply:
Yes, that's correct, but that's not the way this chart is set up.

The questions for this service (Traefik) don't allow for ingressClass to be anything except two booleans `enabled` and `isDefaultClass`.

https://github.com/truecharts/apps/blob/5dd30a1daaeda88e2458bdd0e2f3b406d60824fd/charts/core/traefik/questions.yaml#L106-L124

So the way that this is dealt with in other places in this chart is to use the `.Release.name`

https://github.com/truecharts/apps/blob/5dd30a1daaeda88e2458bdd0e2f3b406d60824fd/charts/core/traefik/templates/_ingressclass.tpl#L20

I can clarify the questions.yaml file with more details about what the ingressClass will be so users are not confused.  Would that be a good solution?

Reply:
Fine